test/transportSpec.js

```diff
@@ -0,0 +1,279 @@
+      })
+    })
+
+    describe('PSK', function () {
+      it('should create packet and condition', function () {
+        // one field name is different
+        this.params.sharedSecret = this.params.secret
+
+        const result = ILP.PSK.createPacketAndCondition(this.params)
+        this.validate(result)
+      })
+    })
+  })
+
+  beforeEach(function () {
+    this.plugin = new MockPlugin()
+  })
+
+  describe('listen', function () {
+    beforeEach(function () {
+      this.params = {
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        secret: Buffer.from('shh_its_a_secret', 'base64')
+      }
+    })
+
+    it('should listen', function () {
+      const res = Transport.listen(this.plugin, this.params, () => {})
+      assert.isFunction(res, 'should return a function')
+    })
+
+    it('should remove listeners with its function', function () {
+      const res = Transport.listen(this.plugin, this.params, () => {})
+      assert.equal(this.plugin.listenerCount('incoming_prepare'), 1)
+      res()
+      assert.equal(this.plugin.listenerCount('incoming_prepare'), 0)
+    })
+
+    describe('IPR', function () {
+      it('should listen via function in IPR', function () {
+        const res = ILP.IPR.listen(this.plugin, this.params, () => {})
+        assert.isFunction(res, 'should return a function')
+      })
+    })
+
+    describe('PSK', function () {
+      it('should listen via function in PSK', function () {
+        const res = ILP.PSK.listen(this.plugin, this.params, () => {})
+        assert.isFunction(res, 'should return a function')
+      })
+    })
+  })
+
+  describe('_validateTransfer', function () {
+    beforeEach(function () {
+      const { packet, condition } = Transport.createPacketAndCondition({
+        destinationAmount: '1',
+        destinationAccount: 'test.example.alice',
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+        data: { foo: 'bar' },
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        expiresAt: moment().add(1, 'seconds').format(),
+        protocol: 'ipr'
+      })
+
+      this.params = {
+        protocol: 'ipr',
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        plugin: this.plugin,
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+        transfer: {
+          id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+          amount: '1',
+          to: 'test.example.alice',
+          from: 'test.example.connie',
+          executionCondition: condition,
+          data: packet
+        }
+      }
+    })
+
+    it('should accept a valid transfer', function * () {
+      yield Transport._validateTransfer(this.params)
+    })
+
+    it('should not accept transfer without condition', function * () {
+      delete this.params.transfer.executionCondition
+      yield expect(Transport._validateTransfer(this.params))
+        .to.be.rejectedWith(/no-execution/)
+    })
+
+    it('should not accept transfer for other account', function * () {
```

Comment:
These tests should say "ignores" instead of "not accept" in cases where it's not rejecting the transfer